src/query/shorthand.ts

```diff
@@ -121,6 +121,13 @@ export function fieldDef(encQ: EncodingQuery,
         value: value(encQ.bin['maxbins'], replace[Property.BIN_MAXBINS])
       });
     }
+  } else if (include[Property.SORT] && encQ.sort && !isEnumSpec(encQ.sort)) {
+    if (!(encQ.sort as SortFieldQuery)['op']) {
```

Comment:
To keep the structure consistent, the right location for this case is below 
```
else if (encQ[nestedPropParent] === false || encQ[nestedPropParent] === null) {
```
case

There you can check if `encQ[nestedPropParent]` is string.   (And add comment that for `sort`, it can be a string.)

This way, `sort` is always handled as a `nestedPropParent` rather than having two different code parts.